plugin/pkg/shoot/validator/admission.go

```diff
@@ -638,7 +638,11 @@ func (c *validationContext) addMetadataAnnotations(a admission.Attributes) {
 		addDNSRecordDeploymentTasks(c.shoot)
 	}
 
-	if !reflect.DeepEqual(c.oldShoot.Spec.Provider.InfrastructureConfig, c.shoot.Spec.Provider.InfrastructureConfig) {
+	// We rely that SSHAccess is defaulted in the shoot creation, that is why we do not check for nils for the new shoot object.
+	if !reflect.DeepEqual(c.oldShoot.Spec.Provider.InfrastructureConfig, c.shoot.Spec.Provider.InfrastructureConfig) ||
+		(c.oldShoot.Spec.Provider.WorkersSettings != nil &&
+			c.oldShoot.Spec.Provider.WorkersSettings.SSHAccess != nil &&
+			c.oldShoot.Spec.Provider.WorkersSettings.SSHAccess.Enabled != c.shoot.Spec.Provider.WorkersSettings.SSHAccess.Enabled) {
```

Comment:
Instead of munging this into the existing statement, can you create a dedicated one (like L653-L656) to improve readability?